src/renderer/viz/expressions/RampImage.js

```diff
@@ -0,0 +1,88 @@
+import defaultSVGs from '../defaultSVGs';
+import { checkType, checkInstance } from './utils';
+import Property from './basic/property';
+import CategoryIndex from './CategoryIndex';
+import ListImage from './ListImage';
+import SVG from './SVG';
+import Base from './base';
+import { OTHERS_GLSL_VALUE, DEFAULT_OPTIONS, DEFAULT_OTHERS } from './constants';
+
+export default class RampImage extends Base {
+    _bindMetadata (metadata) {
+        Base.prototype._bindMetadata.call(this, metadata);
+        this.type = this.palette.childType;
+
+        if (this.others === DEFAULT_OTHERS) {
+            this.others = new SVG(defaultSVGs.circle);
+        } else {
+            checkType('ramp', 'others', 2, 'image', this.others);
+        }
+
+        if (this.input.isA(Property)) {
+            this.input = new CategoryIndex(this.input);
+            this.input._bindMetadata(metadata);
+        }
+
+        checkType('ramp', 'input', 0, 'category', this.input);
```

Comment:
This should be moved above, it doesn't make sense to check the third argument before the first two